### DISCUSSION TITLE
[@types/react] - Exported variable 'DrawerHeaderStyled' has or is using name 'UNDEFINED_VOID_ONLY' from external module "XXX/node_modules/@types/react/canary" but cannot be named.

### DISCUSSION THREAD

**Discussion by trkas** [DefinitelyTyped/DefinitelyTyped#69389] [Issues with a @types package]
Hi,

I have a problem with using @types/react.

I have a Next.js project that I publish to npm. I have a simple Component that I export from the library.

I have following styled Material component:

`// material-ui
import { styled, Theme } from "@mui/material/styles";
import { Box } from "@mui/material";

// ==============================|| DRAWER HEADER - STYLED ||============================== //

interface Props {
  theme: Theme;
  open: boolean;
}

const DrawerHeaderStyled = styled(Box, {
  shouldForwardProp: (prop) => prop !== "open",
})(({ theme, open }: Props) => ({
  ...theme.mixins.toolbar,
  display: "flex",
  alignItems: "center",
  justifyContent: open ? "flex-start" : "center",
  paddingLeft: theme.spacing(open ? 3 : 0),
}));

export default DrawerHeaderStyled;`

When i run npm run build it comes with error :

> ./app/layout/MainLayout/Drawer/DrawerHeader/DrawerHeaderStyled.ts:13:7 Type error: Exported variable 'DrawerHeaderStyled' has or is using name 'UNDEFINED_VOID_ONLY' from external module "/Users/trkas/VSCodeProjects/ng-frontend/node_modules/@types/react/canary" but cannot be named.

I have found out that using @types/react version before 18.2.69 doesn't produce this error, so any ideas on how to avoid this error with latest version(s)?

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
Should be fixed by #69341 I guess.

**Comment by rafalnawojczyk** [comment]
The PR which @codershiba mentioned here should fix this issue, but that type shouldn't really be used outside of types package, as it's internal type. 

If any of you encounters the same problem - I will wrote what helped us. We've had this error only one one file - it was a react custom hook that was using some types. 
Problem was that the type which he was using was extending `React.FC<>`. Apparently you cannot pass React.FC types to custom hook as it produces this error.

**Reply by drewbrend** [reply]
I also ran into this, thanks @rafalnawojczyk for pointing me in the right direction, it was also with a custom hook.

In my case it was using the `Ref<HTMLLIElement>` type from React to type a ref callback that was returned as part of the custom hook.  I switched it to this and the issue was resolved:

```
const ref = (element: HTMLLIElement) => { ... };
```

**Reply by panuhorsmalahti** [reply]
I'm having the same issue when using RefCallback type from the package. My workaround is to create a type similar to RefCallback.
